packages/react-reconciler/src/ReactStrictModeWarnings.js

```diff
@@ -164,11 +170,22 @@ if (__DEV__) {
 
       lowPriorityWarning(
         false,
-        'componentWillReceiveProps is deprecated and will be removed in the next major version. ' +
-          'Use static getDerivedStateFromProps instead.' +
-          '\n\nPlease update the following components: %s' +
-          '\n\nLearn more about this warning here:' +
-          '\nhttps://fb.me/react-async-component-lifecycle-hooks',
+        'componentWillReceiveProps has been renamed to UNSAFE_componentWillReceiveProps, ' +
+          "and the old name won't work in the next major version of React.\n" +
+          'We suggest doing one of the following:\n' +
+          "- If you're updating state whenever props change, " +
+          'move this logic into static getDerivedStateFromProps.\n' +
+          '- If you fetch data or perform other side effects in componentWillReceiveProps, ' +
+          'move this logic into componentDidUpdate.\n' +
+          '- Refactor your code to not use derived state at all, as described at ' +
```

Comment:
Maybe something like?

> Refactor your code to use memoization techniques instead of derived state.
> Learn more at: https://fb.me/react-derived-state